### ISSUE TITLE
`Redis.current=` is deprecated and will be removed in 5.0. (called from: /usr/local/bundle/gems/key_cache-0.0.4/lib/key_cache/cache.rb:64:in `cache')

### ISSUE THREAD

**Issue by cjschooley** [GleasonTech/key_cache#4]

**Comment by joedarc** [comment]
Looks like the alternative is to just set a method for it or set global.

```
  def redis
    @redis ||= Redis.new(url: REDIS_URL)
  end
```

or

```
$redis = Redis.new(url: REDIS_URL)
```

Regarding global

> If your code uses threads, or if you're using a multi-threadeded web server like Puma, sharing $redis that way means all threads use a single Redis connection and [must wait for each other](https://github.com/redis/redis-rb/issues/826) .


Info from [this link](https://makandracards.com/makandra/510011-version-5-of-the-ruby-redis-gem-removes-redis-current), I also don't mind [the extension](https://makandracards.com/makandra/510011-version-5-of-the-ruby-redis-gem-removes-redis-current#section-i-don-t-want-to-close-the-connection-manually-every-time).

**Comment by nwwatson** [comment]
You could also do this using a constant:

```ruby
REDIS_CLIENT = if ENV["REDIS_CLUSTER_URL"]
  Redis.new cluster: ENV["REDIS_CLUSTER_URL"].split(","),
            ssl: ENV.fetch("REDIS_SSL_ENABLED", false)
else
  Redis.new url: ENV["REDIS_URL"]
end
```

**Comment by joedarc** [comment]
> You could also do this using a constant:
> 
> ```ruby
> REDIS_CLIENT = if ENV["REDIS_CLUSTER_URL"]
>   Redis.new cluster: ENV["REDIS_CLUSTER_URL"].split(","),
>             ssl: ENV.fetch("REDIS_SSL_ENABLED", false)
> else
>   Redis.new url: ENV["REDIS_URL"]
> end
> ```

Yeah true, I kind of like the extension the article mentions since it cleans up the connection. We found some locking slowness with Redis last year so anything that helps against that is preferred haha. Either way works though.

**Comment by nwwatson** [comment]
You may also look at moving to [Kredis](https://github.com/rails/kredis), which is developed by the Rails team.
